tests/validator/validator.cpp

```diff
@@ -334,32 +334,71 @@ TEST(Validator, importComponents)
     v.validateModel(m);
     EXPECT_EQ(size_t(0), v.errorCount());
 
-    // Invalid component import- missing refs
+    // Another valid component import
     libcellml::ImportSourcePtr imp2 = std::make_shared<libcellml::ImportSource>();
+    imp2->setUrl("yet-another-other-model.xml");
     libcellml::ComponentPtr importedComponent2 = std::make_shared<libcellml::Component>();
-    importedComponent2->setName("invalid_imported_component_in_this_model");
-    importedComponent2->setSourceComponent(imp2, "");
+    importedComponent2->setName("another_valid_imported_component_in_this_model");
+    importedComponent2->setSourceComponent(imp2, "new_shiny_component_ref");
     m->addComponent(importedComponent2);
     v.validateModel(m);
-    EXPECT_EQ(size_t(3), v.errorCount());
+    EXPECT_EQ(size_t(0), v.errorCount());
 
-    // Invalid component import - duplicate refs
+    // Invalid component import - missing refs
     libcellml::ImportSourcePtr imp3 = std::make_shared<libcellml::ImportSource>();
-    imp3->setUrl("some-other-model.xml");
     libcellml::ComponentPtr importedComponent3 = std::make_shared<libcellml::Component>();
-    importedComponent3->setName("duplicate_imported_component_in_this_model");
-    importedComponent3->setSourceComponent(imp3, "component_in_that_model");
+    importedComponent3->setName("invalid_imported_component_in_this_model");
+    importedComponent3->setSourceComponent(imp3, "");
     m->addComponent(importedComponent3);
     v.validateModel(m);
-    EXPECT_EQ(size_t(4), v.errorCount());
+    EXPECT_EQ(size_t(3), v.errorCount());
 
-    // Invalid component import - unnamed component
+    // Invalid component import - duplicate refs
     libcellml::ImportSourcePtr imp4 = std::make_shared<libcellml::ImportSource>();
-    imp4->setUrl("some-other-different-model.xml");
+    imp4->setUrl("some-other-model.xml");
     libcellml::ComponentPtr importedComponent4 = std::make_shared<libcellml::Component>();
+    importedComponent4->setName("duplicate_imported_component_in_this_model");
     importedComponent4->setSourceComponent(imp4, "component_in_that_model");
     m->addComponent(importedComponent4);
     v.validateModel(m);
+    EXPECT_EQ(size_t(3), v.errorCount());
+
+    libcellml::ImportSourcePtr imp5 = std::make_shared<libcellml::ImportSource>();
+    imp5->setUrl("some-other-different-model.xml");
+    libcellml::ComponentPtr importedComponent5 = std::make_shared<libcellml::Component>();
+    importedComponent5->setSourceComponent(imp5, "component_in_that_model");
+    m->addComponent(importedComponent5);
+    v.validateModel(m);
+    EXPECT_EQ(size_t(5), v.errorCount());
+
+    // Invalid: duplicating component_ref and source
```

Comment:
This one is also valid, right?

Reply:
Yes, it's valid.  Same issue as above where I forgot to update comments when fixing the validator.  Done now :)